src/koopman/koopman.jl

```diff
@@ -0,0 +1,200 @@
+mutable struct Koopman{T, FP, FB, FC} <: abstractKoopmanOperator
+
+    A::AbstractArray{T,2} # Approximation of the operator, Transition
+
+    ψ::FP # Map from states to observable
+    B::FB # Input mapping
+    C::FC # Output mapping
+
+    λ::AbstractArray{Complex{T},1} # Eigenvalues (discrete time)
+    ω::AbstractArray{Complex{T},1} # Frequencies
+    ϕ::AbstractArray{Complex{T},2} # Modes
+
+    # For online DMD
+    Qₖ::AbstractArray{T,2}
+    Pₖ::AbstractArray{T,2}
+
+    # Information
+    islifted::Bool
+    iscontrolled::Bool
+end
+
+
+# Purely linear system
+function Koopman(A::AbstractArray{T,2}; Q::AbstractArray{T,2} = Array{T}(undef, 0, 0), P::AbstractArray{T,2} = Array{T}(undef, 0, 0), dt::R = 0.0) where {T <: Real, R <: Real}
+    # Eigen Decomposition for solution
+    Λ, W = eigen(A)
+
+    if dt > 0.0
+        # Casting Complex enforces results
+        ω = log.(Complex.(Λ)) / dt
+    else
+        ω = eltype(A)[]
+    end
+
+    ψ(u, p, t) = identity(u)
+    C(u, p, t) = identity(u)
+    B(u, p, t) = zeros(u)
+
+    return Koopman(A, ψ, B, C, Complex.(Λ), Complex.(ω), Complex.(W), Q, P, false, false)
+end
+
+# Autonomos
```

Comment:
sp

Reply:
I ended up having sometimes errors thrown due to  eigendecompositions of matrices and the logartihm. To enforce results in this case ( because there should be one ), I cast the eigenvalues to be`Complex` . This is also consistent with most of system analysis. We could always just let it be, but could run into trouble for continouos time generators.